src/openrct2/scripting/ScriptEngine.cpp

```diff
@@ -754,6 +754,12 @@ std::unique_ptr<GameActionResult> ScriptEngine::DukToGameActionResult(const DukV
     result->ErrorTitle = AsOrDefault<std::string>(d["errorTitle"]);
     result->ErrorMessage = AsOrDefault<std::string>(d["errorMessage"]);
     result->Cost = AsOrDefault<int32_t>(d["cost"]);
+
+    auto expenditureType = AsOrDefault<std::string>(d["expenditureType"]);
+    if (!expenditureType.empty())
+    {
+        result->Expenditure = StringToExpenditureType(expenditureType);
```

Comment:
This does not check if StringToExpenditureType returns `ExpenditureType::Count`. (I wonder if it should return that in the first place.)

Reply:
In the bindings I have been using `null` or not defining it for non-applicable values.